gcloud/datastore/query.py

```diff
@@ -66,15 +69,22 @@ class Query(_implicit_environ._DatastoreBase):
     """Mapping of operator strings and their protobuf equivalents."""
 
     def __init__(self,
+                 dataset_id=None,
                  kind=None,
-                 dataset=None,
                  namespace=None,
                  ancestor=None,
                  filters=(),
                  projection=(),
                  order=(),
                  group_by=()):
-        super(Query, self).__init__(dataset=dataset)
+
+        if dataset_id is None:
```

Comment:
I've been doing
```python
dataset_id = dataset_id or _implicit_environ.DATASET_ID
```

I don't have a preference between the two but do prefer we be consistent.

Since this way is more explicit / safe, I should probably go through and clean those up.

(I actually previously had an issue with a `dict` subclass which was `False`-y.)